apps/ios/Shared/Views/ChatList/ChatListView.swift

```diff
@@ -181,10 +211,16 @@ struct ChatListView: View {
 
     private func filteredChats() -> [Chat] {
         let s = searchText.trimmingCharacters(in: .whitespaces).localizedLowercase
-        return s == ""
+        return s == "" && !showFavorites && !showUnread
             ? chatModel.chats
             : chatModel.chats.filter { chat in
-                let contains = chat.chatInfo.chatViewName.localizedLowercase.contains(s)
+                let contains = s != ""
+                                ? chat.chatInfo.chatViewName.localizedLowercase.contains(s)
+                                : (
+                                    (showFavorites && (chat.chatInfo.chatSettings?.favorite ?? false)) ||
+                                    (showUnread && (chat.chatStats.unreadCount > 0 || chat.chatStats.unreadChat)) ||
```

Comment:
It seems if both are toggled, unread but not favorite, and favorite but not unread chats will be unioned. For these acting like filters, it'd be more natural if they produced intersection of favorite unread chats. Union doesn't have much use since filters can be searched separately